test/connectors/CachingSelfAssessmentODSConnectorSpec.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import cats.data.EitherT
+import cats.implicits._
+import org.mockito.ArgumentMatchers.any
+import play.api.Application
+import play.api.inject.bind
+import play.api.mvc.AnyContentAsEmpty
+import play.api.test.FakeRequest
+import repositories.SessionCacheRepository
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, UpstreamErrorResponse}
+import uk.gov.hmrc.mongo.cache.DataKey
+import utils.{BaseSpec, WireMockHelper}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CachingSelfAssessmentODSConnectorSpec extends BaseSpec with ConnectorSpec with WireMockHelper {
+
+  val mockSelfAssessmentODSConnector: SelfAssessmentODSConnector = mock[SelfAssessmentODSConnector]
+  val mockSessionCacheRepository: SessionCacheRepository         = mock[SessionCacheRepository]
+
+  override implicit val hc: HeaderCarrier         = HeaderCarrier()
+  override implicit lazy val ec: ExecutionContext = scala.concurrent.ExecutionContext.global
```

Comment:
Could you inject the ec instead of import global? Doesn't really matter as it's in the tests, though it may slow down the execution of the tests due to the global allocating more threads to counteract blocking actions